### PR TITLE
Make findAuthorByUsername dynamic

### PR DESCRIPTION
**Pull request by nickhartdev**
#### What's this PR do?
- fixes login api calls to take a single username key as opposed to a name / email
#### Where should the reviewer start?
server-helpers.js, the primary function that's changed is findAuthorByUsername
#### How should this be manually tested?
Test the `/api/v1/authors/login/` endpoint with a POST request that has only an email and password in its body, formatted like so:
```
{
  "username": "sometestusername@yahoo.com",
  "password": "somepassword"
}
```
this will work provided there's some test users seeded in the database.
#### What are the relevant tickets?
None!
#### Screenshots (if appropriate)
cat_spin

### CODE REVIEW

**Review by GreysonElkins — APPROVED** [review]
This all looks great! Thanks for doing this!

**Review by carlymclift — APPROVED** [review]
This will be really helpful!

**Comment by GreysonElkins** [comment]
@nickhartdev For some reason I made a new user and it skipped from ID # 5 to #24, is this linked to story ids now or something?

File: server-helpers.js [hunk 1]
```diff
@@ -37,14 +37,14 @@ class UserHelper {
 
   static checkLoginInfo = (info) => {
     const infoProvided = Object.keys(info)
-    const result = { hasEnoughInfo: false, message: '', userType: ''}
+    const result = { hasEnoughInfo: false, message: '', username: '' }
     if (!infoProvided.includes("password")) {
       result.message = 'You need to provide a password'
-    } else if (!infoProvided.some(info => ["email", "name"].includes(info))) {
+    } else if (infoProvided.length === 0 && infoProvided[0] === 'password') {
```
Comment (GreysonElkins) [inline server-helpers.js:43]: Is it save to assume the `username`? will always be the first info? What if there's extra info?

**Review by GreysonElkins — APPROVED** [review]

File: server-helpers.js (hunk 1 above)
Comment (Abdeboskey) [reply server-helpers.js:43]: I'm a little confused by this line, if `infoProvided.length === 0` how can there be an `infoProvided[0]`?

File: server-helpers.js [hunk 2]
```diff
@@ -61,19 +61,22 @@ class UserHelper {
     }
   }
 
-  static findAuthorByNameOrEmail = async (queryType, queryValue) => {
+  static findAuthorByUsername = async username => {
+    let usernameType
+    usernameType = username.includes('@') ? 'email' : 'name'
+
```
Comment (Abdeboskey) [inline server-helpers.js:67]: DOPE

File: server-helpers.js [hunk 3]
```diff
@@ -61,19 +61,22 @@ class UserHelper {
...
     return knex('authors')
       .groupBy('id')
       .select()
-      .having(queryType, '=', queryValue)
+      .having(usernameType, '=', username)
   }
```
Comment (Abdeboskey) [inline server-helpers.js:73]: I need to learn more about the knex SQL methods

**Review by Abdeboskey — APPROVED** [review]
This is rad, all of this back end logic is really fun and y'all have done a great job. Thank you!

**Pull request merged**